lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,535 @@
+
+    @_ssm_retry
+    def _file_transport_command(self, in_path, out_path, ssm_action):
+        ''' transfer a file from using an intermediate S3 bucket '''
+
+        s3_path = out_path.replace('\\', '/')
+        bucket_url = 's3://%s/%s' % (self.get_option('bucket_name'), s3_path)
+
+        if self.is_windows:
+            put_command = "curl -Method PUT -InFile '%s' -Uri '%s'" % (in_path, self._get_url('put_object', self.get_option('bucket_name'), s3_path, 'PUT'))
+        else:
+            put_command = "curl --request PUT --upload-file '%s' '%s'" % (in_path, self._get_url('put_object', self.get_option('bucket_name'), s3_path, 'PUT'))
+
+        get_command = "curl '%s' -o '%s'" % (self._get_url('get_object', self.get_option('bucket_name'), s3_path, 'GET'), out_path)
+
+        client = boto3.client('s3')
+        if ssm_action == 'get':
+            (returncode, stdout, stderr) = self.exec_command(put_command, in_data=None, sudoable=False)
+            with open(out_path.encode('utf-8'), 'wb') as data:
+                client.download_fileobj(self.get_option('bucket_name'), s3_path, data)
+        else:
+            with open(in_path.encode('utf-8'), 'rb') as data:
+                client.upload_fileobj(data, self.get_option('bucket_name'), s3_path)
+            (returncode, stdout, stderr) = self.exec_command(get_command, in_data=None, sudoable=False)
+
+        # Check the return code
+        if returncode == 0:
+            return (returncode, stdout, stderr)
+        else:
+            raise AnsibleError("failed to transfer file to %s %s:\n%s\n%s" %
+                               (to_native(in_path), to_native(out_path), to_native(stdout), to_native(stderr)))
+
+    def put_file(self, in_path, out_path):
+        ''' transfer a file from local to remote '''
+
+        super(Connection, self).put_file(in_path, out_path)
+
+        display.vvv(u"PUT {0} TO {1}".format(in_path, out_path), host=self.host)
+        if not os.path.exists(to_bytes(in_path, errors='surrogate_or_strict')):
+            raise AnsibleFileNotFound("file or module does not exist: {0}".format(to_native(in_path)))
+
+        return self._file_transport_command(in_path, out_path, 'put')
+
+    def fetch_file(self, in_path, out_path):
+        ''' fetch a file from remote to local '''
+
+        super(Connection, self).fetch_file(in_path, out_path)
+
+        display.vvv(u"FETCH {0} TO {1}".format(in_path, out_path), host=self.host)
+        return self._file_transport_command(in_path, out_path, 'get')
+
+    def close(self):
+        ''' terminate the connection '''
+        if self._session_id:
+
+            display.vvv(u"CLOSING SSM CONNECTION TO: {0}".format(self.instance_id), host=self.host)
+            if self._timeout:
+                self._session.terminate()
+            else:
+                cmd = "\nexit\n"
```

Comment:
Remove the PY3 conditional below and just do `cmd = b"\nexit\n"`. There's no variable substitution so just create the byte string itself.

Reply:
Updated in commit https://github.com/ansible/ansible/commit/8a5f982ce77b02d747c2f1954c3e413a449211f8